react/src/components/masthead/components/SprkMastheadAccordionItem/SprkMastheadAccordionItem.js

```diff
@@ -38,11 +38,11 @@ class SprkMastheadAccordionItem extends Component {
       leadingIcon,
       subNavLinks,
       text,
-      itemId,
       ...rest
     } = this.props;
     const { isOpen, height, subNavLinks: stateLinks } = this.state;
     const TagName = element;
+    const itemId = rest.itemId ? rest.itemId : uniqueId('sprk_accordion_item_');
```

Comment:
@Kalovelo Could you walk me through the advantages of turning `itemId` into a rest parameter instead of the typical way of declaring it as a prop then checking for a value?

Reply:
Thank you for your comments!

Since it it used 
here 
https://github.com/sparkdesignsystem/spark-design-system/blob/79e495d9bfb3cb03fd8f3ff70233bde68e95378f/react/src/components/accordions/components/SprkAccordionItem/SprkAccordionItem.js#L68
and here
https://github.com/sparkdesignsystem/spark-design-system/blob/79e495d9bfb3cb03fd8f3ff70233bde68e95378f/react/src/components/accordions/components/SprkAccordionItem/SprkAccordionItem.js#L87

Conditionally setting a unique id to those places with `itemId || uniqueId('sprk_accordion_item_')` would end up with 2 different IDs, therefore a new variable with the same checking would keep the pass the same value for each prop.

Instead of creating a new `const`, I removed the `id` deconstruction and kept the original name.

Should I declare it back to prop and create a new variable which imeplements the 'itemId || uniqueId('sprk_accordion_item_')' checking?

Reply:
> Should I declare it back to prop and create a new variable which imeplements the 'itemId || uniqueId('sprk_accordion_item_')' checking?

Yes please, that would better match how we're doing it in the rest of the codebase. 

I am curious if your way has some performance upgrades/ better js convention with it. Maybe we could improve the code further down the line if that's the case!

Reply:
We'll also need tests for these instances. Thank you for your help with this!